### ISSUE TITLE
[@typescript-eslint/strict-boolean-expressions] await Promise<boolean> not accepted in boolean positoin

### ISSUE THREAD

**Issue by joao-paulo-parity** [typescript-eslint/typescript-eslint#3091]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/strict-boolean-expressions": ["error"]
  }
}
```

```TS
// const isDone: () => Promise<boolean>
    if (await isDone()) {
       // ...
    }
```

**Expected Result**

No issues

**Actual Result**

`error  Unexpected value in conditional. A boolean expression is required  @typescript-eslint/strict-boolean-expressions`

**Additional Info**

Tried wrapping the await within parenthesis, although Prettier just removes it. Are they needed, though? `await <expr>` should be, value-wise, considered as `<expr>`.

"Obvious workaround" of assigning the value to a constant first, then using it, is not really pleasant for such common idiom.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.15.1` |
| `@typescript-eslint/parser`        | `4.15.1` |
| `TypeScript`                       | `4.1.5` |
| `ESLint`                           | `7.20.0` |
| `node`                             | `v15.5.0` |

**Comment by joao-paulo-parity** [comment]
Well, it does work like this already. I'm just messing up.

Sorry for the noise and thanks for the work here!
